### ISSUE TITLE
Add Support for Python 3.6’s __init_subclass__()

### ISSUE THREAD

**Issue by sscherfke** [PyCQA/pylint#1275]
### Steps to reproduce
1. Use [this example](https://docs.python.org/3.6/whatsnew/3.6.html#pep-487-simpler-customization-of-class-creation):
  ```python
class PluginBase:
    subclasses = []

    def __init_subclass__(cls, **kwargs):
        super().__init_subclass__(**kwargs)
        cls.subclasses.append(cls)

class Plugin1(PluginBase):
    pass

class Plugin2(PluginBase):
    pass
```
2. Run pylint (current master)

### Current behavior

`pylint: [no-self-argument] Method should have "self" as first argument`

### Expected behavior

Using `cls` as first argument for this method should be fine.

### pylint --version output

pylint 2.0.0
astroid 1.5.0
Python 3.6.0

**Comment by rogalski** [comment]
Looks like it's just need an extra condition [in this if statement](https://github.com/PyCQA/pylint/blob/6712bbe45888a7f26ca5c3e54f45c3956d87cc58/pylint/checkers/classes.py#L1082).

**Comment by PCManticore** [comment]
Interesting, shouldn't ``__init_subclass__`` be a classmethod, something as ``__prepare__``?

**Comment by rogalski** [comment]
@PCManticore [it was rejected](https://www.python.org/dev/peps/pep-0487/#requiring-an-explicit-decorator-on-init-subclass), i personally disagree with motivation, but since it's part of accepted PEP I guess pylint should follow it...

**Comment by PCManticore** [comment]
Yeah, definitely. Not sure if that is the exact place for fixing this, but I might be wrong and can't really check it now.

**Comment by rogalski** [comment]
@PCManticore alternative is to mark all `_init_subclasses_` methods as classmethod in `astroid` when ast is built. May be reasonable, since it actually behaves as a classmethod.

**Comment by PCManticore** [comment]
That sounds good to me, but we should check what other pylint checks a change like this could affect. Might work though.
